### PR TITLE
Split up n300 demo tests into functionality and performance

### PR DESCRIPTION
**Pull request by skhorasganiTT**
### Ticket
#11960 

### Problem description
- Performance benchmarking for single card demos was being done on VMs

### What's changed
- Split up the n300 demo jobs into functionality and performance
- Set cpu governor to performance mode for the performance tests
- Restricted benchmark data uploading to the n300 perf tests

### Checklist
- [x] Post commit CI passes
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

Single-card demo tests: https://github.com/tenstorrent/tt-metal/actions/runs/10600469118

### CODE REVIEW

**Comment by skhorasganiTT** [comment]
@esmalTT With this change, the mamba tests will be removed for n150 and run only on n300 perf so that the benchmarking data is only from n300. Is that okay for now until you separate the n150 test and/or upload only for n300?

File: tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+if [[ -z "$TT_METAL_HOME" ]]; then
+  echo "Must provide TT_METAL_HOME in environment" 1>&2
+  exit 1
+fi
+
+source tests/scripts/single_card/run_demos_single_card_wh_common.sh
+
+# Not working on N150, working on N300
+unset WH_ARCH_YAML
+rm -rf built
```
Comment (ttmchiou) [inline tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh:14]: why do we need to remove the built folder here?

Comment (skhorasganiTT) [reply tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh:14]: I'm not sure about that, it was there from before and I didn't modify it.

**Comment by esmalTT** [comment]
> @esmalTT With this change, the mamba tests will be removed for n150 and run only on n300 perf so that the benchmarking data is only from n300. Is that okay for now until you separate the n150 test and/or upload only for n300?

That’s fine for now. Thanks @skhorasganiTT.

**Review by esmalTT — APPROVED** [review]

**Review by TT-billteng — APPROVED** [review]

File: tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh (hunk 1 above)
Comment (ttmchiou) [reply tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh:14]: can we see if this still works without this?

**Comment by mtairum** [comment]
@skhorasganiTT I don't oppose these changes, but for Mistral-7B you could still keep it in both N150 and N300 since we are now making use of the fixture `is_single_card_n300` -> meaning that functionality runs on both N150 and N300, but only generates the perf metrics if on N300.

File: tests/scripts/single_card/run_demos_single_card_wh_common.sh [hunk 2]
```diff
@@ -12,9 +12,5 @@ fi
 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto --disable-warnings -q -s --input-method=cli --cli-input="YOUR PROMPT GOES HERE!"  models/demos/wormhole/falcon7b/demo_wormhole.py::test_demo[wormhole_b0-True-user_input0-1-default_mode_1024_stochastic]
 # llama3.1-8B
 WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/demos/wormhole/llama31_8b/demo/demo.py --timeout 600
-WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/demos/wormhole/llama31_8b/demo/demo_with_prefill.py --timeout 600
```
Comment (tt-rkim) [inline tests/scripts/single_card/run_demos_single_card_wh_common.sh:15]: This means we no longer some of these models on N150.

For example, it seems the only mamba test we have is now no longer on N150.

What's the policy on this, and do we have buy in from customer team? @uaydonat @dvartaniansTT @mbahnasTT

File: tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh (hunk 1 above)
Comment (TT-billteng) [reply tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh:14]: let's try this experiment before checking in

File: tests/scripts/single_card/run_demos_single_card_wh_common.sh (hunk 2 above)
Comment (skhorasganiTT) [reply tests/scripts/single_card/run_demos_single_card_wh_common.sh:15]: @tt-rkim I added back the n150 tests while keeping the benchmark uploads only for n300 perf

File: tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh (hunk 1 above)
Comment (skhorasganiTT) [reply tests/scripts/single_card/run_demos_single_card_n300_func_tests.sh]: Removed that line and re-tested, passed

**Comment by skhorasganiTT** [comment]
@tt-rkim @TT-billteng @ttmchiou I refactored the single-card demos to be invoked with functions instead of files, similar to the other demo pipelines, since it was becoming too unorganized. Could you please check again? (The updated pipeline run is in the description)

@esmalTT @mtairum I re-added the n150 tests while keeping the measurement uploading restricted to n300-perf, so there's no more action for you @esmalTT, and @mtairum you can safely remove the use of the fixture in the demo when this goes in.

**Review by tt-rkim — APPROVED** [review]
ME GUSTA

please check all the necessary tests are actually run on each branch of the single card demos

**Review by ttmchiou — APPROVED** [review]
looks good, just make sure tests pass before merging as per @tt-rkim's comment

**Comment by mbahnasTT** [comment]
@tt-rkim can you please elaborate on the situation of n150? We will be selling this card, and we care about its functionality and performance. We don't want to repeat the same situation of selling e75 but not fully supporting it.

CC: @uaydonat @dvartaniansTT @milank94

**Comment by tt-rkim** [comment]
@uaydonat and his team maintain functionality for both N150 and N300 for these demo models, unless clearly stated on the README.

For performance, we only track single-card model performance for the demo models on N300.

@uaydonat can clarify any additional questions you have.

**Pull request merged**